### PR TITLE
Add the possibility to set the maximum number of headers

### PR DESCRIPTION
**Pull request by 3kyro**
kontrakcja PR https://github.com/scrive/kontrakcja/pull/7076

### CODE REVIEW

File: http-conduit/Network/HTTP/Simple.hs [hunk 1]
```diff
@@ -110,7 +110,6 @@ import qualified Data.Aeson as A
 
 import qualified Data.Traversable as T
 import Control.Exception (throw, throwIO, Exception)
-import Data.Monoid
```
Comment (jonathanjouty) [inline http-conduit/Network/HTTP/Simple.hs:113]: Irrelevant change, we should probably remove this especially if we want to upstream our changes?

File: http-client/Network/HTTP/Client/Types.hs [hunk 2]
```diff
@@ -906,4 +919,12 @@ data StreamFileStatus = StreamFileStatus
 newtype MaxHeaderLength = MaxHeaderLength
     { unMaxHeaderLength :: Int
     }
-    deriving (Eq, Show)
+    deriving (Eq, Show, Ord, T.Typeable)
+
+-- | The maximum number of header lines.
+--
+-- @since TODO
```
Comment (jonathanjouty) [inline http-client/Network/HTTP/Client/Types.hs:926]: We would need to fix these TODO documentation comments before submitting a PR upstream

**Review by jonathanjouty — APPROVED** [review]
Implementation looks fine, I'm not sure why GitHub Actions have not run, despite the upstream actions involving a lot of failures, there are several that pass so it would be nice to at least get one ✅

**Comment by jonathanjouty** [comment]
Right, I had to manually enable actions as this is a forked repository and it could steal something from the Scrive org.
Manual run: https://github.com/scrive/http-client/actions/runs/9097895860

**Review by jonathanjouty — COMMENTED** [review]
I actually propose to not merge this PR until it is fully ready to be made into an upstream PR, as we can set the `source-repository-package.tag` in our internal usages regardless.

**Comment by marinelli** [comment]
We can close this PR, we have a new release of `http-client` with the required changes.

**Pull request closed without merging**